packages/core/src/utilities/getCurrentVolumeViewportSlice.ts

```diff
@@ -0,0 +1,168 @@
+import { glMatrix, mat4, vec3 } from 'gl-matrix';
+import { transformIndexToWorld, transformWorldToIndex } from '.';
+import { IVolumeViewport, Point2, Point3 } from '../types';
+
+/**
+ * Get the image data for the current slice rendered on the viewport.
+ * It does not work for oblique views.
+ * @param viewport - Volume viewport
+ * @returns Slice image dataand  matrices to convert from volume
+ *   to slice and vice-versa
+ */
+function getCurrentVolumeViewportSlice(viewport: IVolumeViewport) {
+  const {
+    dimensions,
+    scalarData,
+    imageData: vtkImageData,
+  } = viewport.getImageData();
+
+  // Convert coordinates from index to canvas space
+  const ijkToCanvas = (volPoint: Point3) => {
+    const worldPoint = transformIndexToWorld(vtkImageData, volPoint);
+
+    return viewport.worldToCanvas(worldPoint);
+  };
+
+  // Convert coordinates from canvas to index space
+  const canvasToIJK = (canvasPoint: Point2) => {
+    const worldPoint = viewport.canvasToWorld(canvasPoint);
+    return transformWorldToIndex(vtkImageData, worldPoint);
+  };
```

Comment:
extract these to their own utilities since they are useful. Something like transformIJKToCanvas